File: utils/utils-base/src/com/xenoage/utils/font/TextMetrics.java
package com.xenoage.utils.font;

/**
 * Metrics of text measured with a {@link TextMeasurer}.
 * 
 * @author Andreas Wenger
 */
public class TextMetrics {
	
	private float ascent, descent, leading, width;
	

	public TextMetrics(float ascent, float descent, float leading, float width) {
		this.ascent = ascent;
		this.descent = descent;
		this.leading = leading;
		this.width = width;
	}

	/**
	 * Gets the ascent of this text in mm.
	 * This is the height of the text above the baseline.
	 */
	public float getAscent() {
		return ascent;
	}

	/**
	 * Gets the descent of this font in mm.
	 * This is the height of the paragraph below the baseline.
	 */
	public float getDescent() {
		return descent;
	}

	/**
	 * Gets the leading of this font in mm.
	 * This is the height between the bottommost point of this text
	 * to the topmost point of the following line of text
	 */
	public float getLeading() {
		return leading;
	}

	/**
	 * Gets the width of this text in mm.
	 */
	public float getWidth()  {
		return width;
	}

}

File: utils-common/src/com/xenoage/utils/font/TextMetrics.kt
package com.xenoage.utils.font

import com.xenoage.utils.Mm

/**
 * Metrics of text measured with a [TextMeasurer].
 */
class TextMetrics(

		/** The ascent of this text in mm. This is the height of the text above the baseline. */
		val ascent: Mm,

		/** The descent of this font in mm. This is the height of the paragraph below the baseline. */
		val descent: Mm,

		/**
		 * The leading of this font in mm.
		 * This is the height between the bottommost point of this text
		 * to the topmost point of the following line of text
		 */
		val leading: Mm,

		/** The width of this text in mm. */
		val width: Mm
)
